src/main/java/ch/fhnw/algd1/converters/binary/BinConverter.java

```diff
@@ -4,12 +4,83 @@ public class BinConverter {
 	public static String toString(int x) {
 		// TODO expect x to be in range [-128, 127], return string with 8 binary
 		// digits representing x in 2-complement
-		return "00000000";
+		String result = "";
+		boolean positive = x>=0;
+		if(positive){
+			while(x>0){
+				if(x%2 == 0){
+					result = "0"+result;
+				} else {
+					result = "1"+result;
+				}
+				x = x/2;
+			}
+			int count = 8-result.length();
+			for(int i = 0; i < count; i++ ){
+				result = "0"+result;
+			}
+		} else{
+			x = Math.abs(x);
+			String intermediateResult = "";
+			while(x>0){
+				if(x%2 == 0){
+					intermediateResult = "0"+intermediateResult;
+				} else {
+					intermediateResult = "1"+intermediateResult;
+				}
+				x = x/2;
+			}
+			int count = 8-intermediateResult.length();
+			for(int i = 0; i < count; i++ ){
+				intermediateResult = "0"+intermediateResult;
+			}
+
+			for(int i = 0; i< intermediateResult.length(); i++){
+				if(intermediateResult.charAt(i) == '1'){
+					result += 0;
+				} else{
+					result += 1;
+				}
+			}
+
+
+			for(int i = result.length()-1; i >= 0; i--){
+				int memory = 0;
+				if(result.charAt(i)=='0'){
+					result = replaceChar(result, '1',i);
+					return result;
+				} else {
+					result = replaceChar(result, '0',i);
+				}
+			}
+		}
+		return result;
 	}
 
 	public static int fromString(String text) {
 		// TODO expect text to contain 8 binary digits, parse to int value in
 		// 2-complement
-		return 0;
+		int result = 0;
+		if(text.charAt(0)=='1'){
+			result -= Math.pow(2,7);
+		}
+		int pow = 6;
+		for(int i = 1; i< 8; i++){
+			if(text.charAt(i) == '1'){
+				result += Math.pow(2, pow);
+			}
+			pow--;
+		}
+		return result;
+	}
+
+	public static String replaceChar(String txt, char c , int i){
+		if(i == 0){
+			return c + txt.substring(1);
+		} else if(i == txt.length()-1){
+			return txt.substring(0,txt.length()-1)+c;
+		} else{
+			return txt.substring(0,i)+c+txt.substring(i+1);
```

Comment:
Wenn ich die Definition der `substring`-Operation richtig in Erinnerung habe, braucht es die Fallunterscheidung nicht, weil die allgemeine Variante unten auch die beiden Randfälle korrekt abdeckt.
Man könnte sich - wenn man das häufiger braucht - auch fragen, ob da jetzt ein StringBuilder oder ein char-Array nicht die bessere Basis wäre. Da könnte man dann einfach ein einzelnes Zeichen überschreiben, ohne alles andere zu kopieren,